sql: TestCancelQueryPermissions failed

sql.TestCancelQueryPermissions [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3859936&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3859936&tab=artifacts#/) on master @ [f64232150885928ce2d7680136f7e0da413ac10c](https://github.com/cockroachdb/cockroach/commits/f64232150885928ce2d7680136f7e0da413ac10c):


```
=== RUN   TestCancelQueryPermissions
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCancelQueryPermissions833194974
    test_log_scope.go:80: use -show-logs to present logs inline
    run_control_test.go:335: expected error 'query execution canceled', got: sql: database is closed
    run_control_test.go:368: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCancelQueryPermissions833194974
--- FAIL: TestCancelQueryPermissions (1.91s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestCancelQueryPermissions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/run_control_test.go
+++ b/pkg/sql/run_control_test.go
@@ -26,7 +26,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sqltestutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -243,7 +242,6 @@ GRANT admin TO has_admin2;
 
 func TestCancelQueryPermissions(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	skip.WithIssue(t, 73544, "flaky test")
 	defer log.Scope(t).Close(t)
 
 	// getQueryIDs retrieves the IDs of any currently running queries for the
@@ -365,6 +363,8 @@ GRANT admin TO has_admin2;
 			}
 		}()
 	}
+	// Give the cancel queries a chance to propagate before stopping the cluster.
+	time.Sleep(time.Second)
 	testCluster.Stopper().Stop(ctx)
 	wg.Wait()
 }

